firecrest/AsyncClient.py

```diff
@@ -777,10 +850,76 @@ async def compress(self, machine: str, source_path: str, target_path: str) -> st
             additional_headers={"X-Machine-Name": machine},
             data={"targetPath": target_path, "sourcePath": source_path},
         )
-        self._json_response([resp], 201)
+        timeout_str = "Command has finished with timeout signal"
+        if (
+            resp.status_code == 201 or
+            fail_on_timeout or
+            resp.status_code != 400 or
+            resp.json()["error"] != timeout_str
+        ):
+            self._json_response([resp], 201)
+        else:
+            logger.debug(
+                f"Compression of {source_path} to {target_path} has finished "
+                f"with timeout signal. Will submit a job to compress the "
+                f"file."
+            )
+            job_info = await self.submit_compress_job(
+                machine,
+                source_path,
+                target_path
+            )
+            jobid = job_info['jobid']
+            active_jobs = await self.poll_active(
+                machine,
+                [jobid]
+            )
+            intervals = itertools.cycle(
+                [5,  # We know it took at least 5 seconds
```

Comment:
Even though you are aware it takes at least 5 second, it's better to have pyfirecrest designed in best ways. So that if Firecrest server improved in future, we don't need to come back here and fix this.